### PR TITLE
Fix error message in push command when bala not found

### LINKED ISSUE #29347: [Local repository support] Incorrect instructions in the error message
It should have been `bal build -c`

```bash
→ bal push --repository=local
ballerina: cannot find bala file for the package: checkpanic_analyzer. Run 'bal build' to compile and generate the bala.
```

### LINKED ISSUE #29348: The "bal push" command does not consider the package version in Balleirna.toml
The `bal push` command should cross-check the `.bala` version in the target directory with the package version in `Ballerina.toml`. 

Here is what I've observed. I did `bal build -c` when my package version was `0.1.0`. Then I changed the package version in `Ballerina.toml` to `0.1.1`. Then I did `bal push --repository=local`, and it successfully pushed the old version. 

I expected the `bal push` command to complain.

### PR DESCRIPTION
**Pull request by azinneera**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by pramodya1994 — DISMISSED** [review]

**Comment by pubudu91** [comment]
Shall we change the branch of this, if this is not meant for alpha3, so that we don't merge this to the release branch by accident?

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java [hunk 1]
```diff
@@ -212,20 +213,29 @@ private static Path validateBala(BuildProject project, CentralAPIClient client)
...
         }
 
         Path packageBalaFile = findBalaFile(pkgName, orgName, balaOutputDir);
         if (null == packageBalaFile) {
             throw new ProjectException("cannot find bala file for the package: " + pkgName + ". Run "
-                    + "'bal build' to compile and generate the bala.");
+                    + "'bal build -c' to compile and generate the bala.");
+        }
+
+        if (!packageBalaFile.toString().endsWith(
```
Comment (pramodya1994) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java:233]: Hi, this is another validation we added to confirm with the Ballerina.toml version and We still send the version taken from bala right?

**Review by pramodya1994 — APPROVED** [review]

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java (hunk 1 above)
Comment (pramodya1994) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java:233]: Approved the PR after checking this.

Comment (azinneera) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java:233]: We don't continue pushing if the validation fails. The versions should be identical to be able to push.

**Pull request merged**